src/storage-client/src/controller.rs

```diff
@@ -2020,8 +2095,24 @@ where
         metadata.collection_metadata.remap_shard = remap_shard;
     }
 
+    /// Closes the identified shards from further reads or writes.
+    ///
+    /// The string accompanying the `ShardId` is the shard's "purpose",
+    /// necessary to open read and write handles to the shard.
     #[allow(dead_code)]
-    async fn truncate_shards(&mut self, shards: &[(ShardId, String)]) {
+    async fn finalize_shards(&mut self, shards: &[(ShardId, String)]) {
```

Comment:
as a followup, we should make this function finalize all the shards concurrently!

Reply:
Done--PTAL. I don't have intuition around the cost of increasing `buffer_unordered` but we don't expect to be finalizing more than a very small number of shards at a time. If, in that case, we don't want the added complexity, no sweat to just revert the commit.

Reply:
Chatted with Aljoscha about this and he suggested because we'll so rarely be finalizing shards better to leave this code a little more readable and just finalizing sequentially. If this is ever a bottleneck, we can re-parallelize it, though.

Reply:
@sploiselle sounds good!